src/core/search/base.cc

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/search/base.h"
+
+namespace dfly::search {
+
+std::string_view QueryParams::operator[](std::string_view name) const {
+  if (auto it = params.find(name); it != params.end())
+    return it->second;
+  return "";
+}
+
+std::string& QueryParams::operator[](std::string_view k) {
+  return params[k];
+}
+
+WrappedStrPtr::WrappedStrPtr(const std::string& s) : ptr{std::make_unique<char[]>(s.size() + 1)} {
+  std::memcpy(ptr.get(), s.c_str(), s.size() + 1);
+}
+
+bool WrappedStrPtr::operator<(const WrappedStrPtr& other) const {
+  return std::strcmp(ptr.get(), other.ptr.get());
```

Comment:
`strcmp` finishes comparison at `\0`. could `other` or `ptr` have `\0` within them?
(I noticed you used `memcpy` rather than `strcpy`, that's why I'm asking)

Reply:
No, I take those are json/hashmap values

Reply:
Why should I use strcpy if I know the size of the string_view?

Reply:
I'm not saying that you should or shouldn't use `strcpy`, just that usually these APIs are not used together. As long as hashmap values are guaranteed not to have `\0` I think we're good, just wanted to make sure.